### PR TITLE
[release-4.12] OCPBUGS-22247,OCPBUGS-19549: Fix empty clusterName references for GenerateMachinePublicIPName

### PR DESCRIPTION
**Pull request by JoelSpeed**
Manual cherry-pick of #52 to fix up the test script which isn't working on the 4.12 branch currently

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,7 +38,9 @@ GOOS ?= $(shell go env GOOS)
 
 NO_DOCKER ?= 0
 
-ifeq ($(shell command -v podman > /dev/null 2>&1 ; echo $$? ), 0)
+ifeq (${IS_CONTAINER}, TRUE)
```
Comment (damdo) [inline Makefile:41]: FYI this is what we did in 4.14/4.13 re this issue: https://github.com/openshift/machine-api-provider-azure/pull/74/files

Comment (JoelSpeed) [reply Makefile:41]: Right, that would have worked but does change the dev experience. Given this is 4.12, I don't think I care much about consistency for this, would rather just get it fixed. The net result is the same 😅

**Review by damdo — COMMENTED** [review]
/lgtm

**Comment by JoelSpeed** [comment]
Adding cherry-pick label since QE have already verified pre-merge

**Pull request merged**